File: app/src/main/java/com/communityx/adapters/EventPagerAdapter.java
package com.communityx.adapters;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import com.communityx.fragments.MyHostedEventFragment;

import java.util.List;

public class EventPagerAdapter extends FragmentPagerAdapter {

    private List<String> mListTitles;
    private Fragment[] fragments = {
            new MyHostedEventFragment(),
            new MyHostedEventFragment()
    };

    public EventPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public void setTitles(List<String> mListTitles) {
        this.mListTitles = mListTitles;
    }

    @Override
    public Fragment getItem(int i) {
        return fragments[i];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mListTitles.get(position);
    }
}

File: app/src/main/java/com/communityx/adapters/EventPagerAdapter.kt
package com.communityx.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.communityx.fragments.MyHostedEventFragment

class EventPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private var mListTitles: List<String>? = null
    private val fragments = arrayOf<Fragment>(MyHostedEventFragment(), MyHostedEventFragment())

    fun setTitles(mListTitles: List<String>) {
        this.mListTitles = mListTitles
    }

    override fun getItem(i: Int): Fragment {
        return fragments[i]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mListTitles!![position]
    }
}
